parser-typechecker/tests/Unison/Test/MCode.hs

```diff
@@ -72,7 +72,7 @@ env m = mapInsert (bit 24) m
 asrt :: Section
 asrt = Ins (Unpack Nothing 0)
      $ Match 0
-     $ Test1 1 (Yield ZArgs)
+     $ Test1 1 (Yield (BArg1 0))
```

Comment:
Can you help me understand this change from `ZArgs` to `BArg1 0`?

Reply:
The tests were all implemented with this trivial function that tests its argument, and either does a zero-argument return or throws an exception. Then the tests just wrap it in something that will be successful if it terminates, and a failure if the exception is thrown. You can think of the zero-argument return as an optimized version of `a -> ()`, where we've optimized out building a boxed `()`.

The top level exception handler expects a value to be percolating up, though, for the default case. So the zero-argument return was making it segfault by reading outside the stack (because nothing was on the stack). So, instead I made the test function just return its boxed input in the successful case (that's the `BArg1 0`). Then the exception handler gets this value. It does nothing with it, but it's necessary to keep it from segfaulting.

Reply:
@dolio thanks that was a super helpful explanation